Fix autoscaling Knative properties

Sgitario opened the pull request:
This PR the following issues:
- The generated `config-autoscaler` configmap does not set the namespace which should always be `knative-serving` (see for example https://knative.dev/docs/serving/autoscaling/scale-to-zero/#enable-scale-to-zero).
- Some properties do add the annotations at Knative service metadata level, and it should add them at Knative service spec template metadata level.
- The property `quarkus.knative.global-auto-scaling.containerConcurrency` (hard limit option) wrongly uses the `config-autoscaler` configmap and it should use `config-defaults` configmap (see https://knative.dev/docs/serving/autoscaling/concurrency/#hard-limit)

Moreover, add the autoscaling Knative properties to the Knative documentation.

Fix https://github.com/quarkusio/quarkus/issues/23786

gsmet commented:
@iocanel could you have a look at this one today? It looks like something we want in 2.7.2. Thanks!

iocanel reviewed (changes requested):
Overall, looks good.
Some test classes need to be aligned with the rest of the tests (we can have some using the 'Test'  suffix and others not). Other than that looks great!

Sgitario commented on integration-tests/kubernetes/quarkus-standard-way/src/test/java/io/quarkus/it/kubernetes/KnativeWithRevisionAutoscaling.java:
```diff
@@ -0,0 +1,58 @@
+import io.quarkus.test.ProdModeTestResults;
+import io.quarkus.test.QuarkusProdModeTest;
+
+public class KnativeWithRevisionAutoscaling {
```
ops, my bad, adding it.

Sgitario committed:
Fix autoscaling Knative properties

This PR the following issues:
- The generated `config-autoscaler` configmap does not set the namespace which should always be `knative-serving` (see for example https://knative.dev/docs/serving/autoscaling/scale-to-zero/#enable-scale-to-zero).
- Some properties do add the annotations at Knative service metadata level, and it should add them at Knative service spec template metadata level.
- The property `quarkus.knative.global-auto-scaling.containerConcurrency` (hard limit option) wrongly uses the `config-autoscaler` configmap and it should use `config-defaults` configmap (see https://knative.dev/docs/serving/autoscaling/concurrency/#hard-limit)

Moreover, add the autoscaling Knative properties to the Knative documentation.

Fix https://github.com/quarkusio/quarkus/issues/23786

Sgitario commented on integration-tests/kubernetes/quarkus-standard-way/src/test/java/io/quarkus/it/kubernetes/KnativeWithGlobalAutoscaling.java:
```diff
@@ -0,0 +1,53 @@
+import io.quarkus.test.ProdModeTestResults;
+import io.quarkus.test.QuarkusProdModeTest;
+
+public class KnativeWithGlobalAutoscaling {
```
Renamed

iocanel reviewed (approved):
Lgtm
